cuda/btx_cudamatching_model.yaml

```diff
@@ -0,0 +1,58 @@
+:environment:
+  :entries:
+  - :name: ^hostname$
+    :type: string
+:stream_classes:
+- :default_clock_class: {}
+  :event_common_context_field_class:
+    :type: structure
+    :members:
+    - :name: ^vpid$
+      :field_class:
+        :type: integer_signed
+    - :name: ^vtid$
+      :field_class:
+        :type: integer_unsigned
+  :event_classes:
+  - :set_id: entries
+    :name: "_entry$"
+
+  - :set_id: exits
+    :register: false
+    :name: "_exit$"
+
+  - :set_id: exits_cudaError_present
+    :domain: exits
+    :payload_field_class:
+      :type: structure
+      :members:
+      - :name: ^cuResult$
+        :field_class:
+          :type: integer_signed
+          :cast_type: ^CUresult$
+
+  - :set_id: exits_cudaError_absent
+    :domain: exits - exits_cudaError_present 
+
+  # Memory traffic
+  - :set_id: traffic
+    :name: "cuMemcpy(2D|3D)?(Unaligned)?(DtoH|DtoD|HtoD)(Aync)?_v2(_ptds|_ptsz)?"
+    :register: false
+  - :set_id: entries_traffic
+    :domain: entries & traffic
+    :payload_field_class:
+      :type: structure
+      :members:
+      - :name: ^ByteCount$
+        :field_class:
+          :cast_type: size_t
+          :type: integer_unsigned
```

Comment:
Does some `ByteCount` are NOT size_t? If not, you can remove the field_class

Reply:
I guess this is a question of failure modes and of documentation. I like being more explicit to document the type in the matching file. But then I guess if there was a ByteCount that was not size_t, the error would not be caught and the event would pass through the filter unchanged?

Reply:
No, it will be a compile-time error. As the `ByteCount` will have two types, hence two function signatures for the callbacks, which is obvious not possible.